Assets/Scripts/Fighting/TestTurret.cs

```diff
@@ -2,23 +2,26 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-public class TestTurret : MonoBehaviour
+namespace Fighting
 {
-    void Start()
+    public class TestTurret : MonoBehaviour
     {
-        StartCoroutine(TurretFire());
-    }
+        void Start()
+        {
+            StartCoroutine(TurretFire());
+        }
 
-    private IEnumerator TurretFire()
-    {
-        while (true) {
-            Debug.DrawRay(transform.position, transform.forward, Color.red);
-            if (Physics.Raycast(transform.position, transform.forward, out var hit, 5f) /* && hit.collider.GetComponent<Hitbox>() != null*/)
+        private IEnumerator TurretFire()
```

Comment:
This should not be an enumerator, I think - but rather a call in update/fixed update with time stamps. But I'm not sure.

Reply:
Its a coroutine, so maybe its fine? Idk about Unity Best Practices with regards to timed actions tho

Reply:
I think it should be fine - but that should not be used more than a few times throughout the project, I could imagine Coroutines being slow

Reply:
@kosch104 I would disagree. Coroutines aren't expensive. They act as normal async routines. The most important thing is to close them which they do by themselves.

See for reference: https://jacksondunstan.com/articles/2981

Reply:
Okay! Then I'll mark this as resolved.